### ISSUE TITLE
Release tarballs should include configure script

### ISSUE THREAD

**Issue by jwakely** [dagargo/elektroid#18]
Conventionally, a release tarball already contains the `configure` script, so that end users can just run `./configure` and not need to have `autoconf` installed (expecting developers working with the source repository to use `autoreconf` is fine, just not users of the release tarballs).

**Comment by jwakely** [comment]
Specifically, running `autoreconf --install && ./configure && make dist-gzip dist-zip` will create `.zip` and `.tar.gz` relases that contain the `configure` script.

**Comment by dagargo** [comment]
I'm not really sure why you are mentioning this. So, since I'm not an expert on how to properly distribute tarballs or autotools, I need to say a few things.

This fails at the end of the `dist-zip` target.
```
autoreconf --install && ./configure && make dist-gzip dist-zip
```
But this works.
```
autoreconf --install && ./configure && make dist-gzip && make dist-zip
```
Not sure if that's a typo or if there is something wrong somewhere.

Regarding the `./configure` script, it looks like it's properly included in both the tarball and the zip since version 0.2.

However, I've noticed that some files where missing in the distribution files. I fixed this in fe74c3ede1db0ddb2eaa5722144fee847a61447c.

Were you mentioning this because of the `README.md`? Am I missing something?

**Comment by dagargo** [comment]
I think I've understood you. Were you referring to the tarballs that GitHub adds automatically for every tag? I didn't know it was possible to overwrite them until now.

If you confirm this I'll fix all the releases as, yes, they're broken in the sense that `autoconf` is needed and they can not be used the standard way.

**Comment by simonvanderveldt** [comment]
@dagargo FYI most of the time releases/release tarballs contain the configure script, so it's not necessary to have autotools installed when building/installing it.
This SO answer gives an IMHO pretty useful explanation about the why as well as the how https://stackoverflow.com/a/41360302

**Comment by jwakely** [comment]
Hi , sorry for the late reply.

> I think I've understood you. Were you referring to the tarballs that GitHub adds automatically for every tag?

Yes, I think what I did was go to the "Releases - 7 tags" link on the https://github.com/dagargo/elektroid/releases project page, which goes to https://github.com/dagargo/elektroid/releases

From there I downloaded a .tar.gz of the 1.2 release, and built it locally. Having downloaded a release tarball, I expected to be able to just run `configure`. But what I didn't realise is that the tarball was just something automatically created by github from the tag in the repo. And so of course it doesn't have `configure`, and it's necessary to run `autoreconf` yourself before configuring.

So this is my misunderstanding about how those release tarballs are prepared. If you'd like to close this issue, that's fine by me. Everything is working as expected.

**Comment by dagargo** [comment]
Let's leave this open for now.

I'll give it a try during the next release, which I haven't quite planned yet.

**Comment by dagargo** [comment]
Now that Elektroid 1.3 is out, I've published [here](https://github.com/dagargo/elektroid/releases/tag/1.3) the `tar.gz`and `zip` files as explained in the [stackoverflow comment above](https://stackoverflow.com/a/41360302).

Let me know if this works for you.

**Comment by jwakely** [comment]
Yes, I can build the RPM from the tarball without running `autoreconf` now, thanks!

I'm seeing the header files installed due to fe74c3ede1db0ddb2eaa5722144fee847a61447c (which doesn't seem useful for users of the application) but that's a separate topic.

**Comment by jwakely** [comment]
Should those headers be listed with [`noinst_HEADERS`](https://www.gnu.org/software/automake/manual/html_node/Headers.html) instead of `include_HEADERS`?

**Comment by dagargo** [comment]
Certainly, we don't need the `include_HEADERS` variable. But we need to include them somehow because otherwise `distcheck` fails.

In the link you mentioned, they say it's recommended to include the headers in the program’s `_SOURCES` variable.

> However when the header actually belongs to a single convenience library or program, we recommend listing it in the program’s or library’s _SOURCES variable (see Program Sources) instead of in noinst_HEADERS.

I've proceeded this way and it's fixed in commit 5579ce3bc57fcfa365f30ab4dd6d9631ff31bfeb.

Let me know if this works for you.

@snd1, I've tested it and looks like it doesn't break anything. Is everything right from the Debian POV?
